frontend/public/components/modals/column-management-modal.tsx

```diff
@@ -0,0 +1,170 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+import * as _ from 'lodash';
+import {
+  Alert,
+  DataList,
+  DataListCheck,
+  DataListItem,
+  DataListItemRow,
+  DataListCell,
+  DataListItemCells,
+} from '@patternfly/react-core';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { setColumnManagementFilter } from '../../actions/ui';
+
+export const MAX_VIEW_COLS = 9;
+
+const OVERRIDES = ['Name'];
+
+const getDataListRows = (
+  rowData: any[],
+  onChange: (checked: boolean, event: any) => void,
+  disableUncheckedRows: boolean,
+): React.ReactNode[] =>
+  rowData.map((column) => (
+    <DataListItem aria-labelledby="table-column-management-item1" key={column.title}>
+      <DataListItemRow>
+        <DataListCheck
+          isDisabled={(disableUncheckedRows && !column.visible) || OVERRIDES.includes(column.title)}
+          aria-labelledby="table-column-management-item1"
+          checked={column.visible}
+          name={column.title}
+          onChange={onChange}
+        />
+        <DataListItemCells
+          dataListCells={[
+            <DataListCell id={`table-column-management-item-${column.title}`} key={column.title}>
+              {column.title}
+            </DataListCell>,
+          ]}
+        />
+      </DataListItemRow>
+    </DataListItem>
+  ));
+
+export const ColumnManagementModal: React.FC<ColumnManagementModalProps> = ({
+  kinds,
+  cancel,
+  close,
+  columnFilters,
+}) => {
+  const dispatch = useDispatch();
+  const initialDefaultColumns = !_.isEmpty(columnFilters)
+    ? _.cloneDeep(columnFilters).filter((column) => column.title.length > 0 && !column.additional)
```

Comment:
No need for `cloneDeep`. `filter` already creates a new array.

```suggestion
    ? columnFilters.filter((column) => column.title.length > 0 && !column.additional)
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
done